examples/with-react-i18next/i18n/hoc.tsx

```diff
@@ -0,0 +1,46 @@
+import i18next from 'i18next'
+import { NextAppContext } from 'next/app'
+import { DefaultQuery } from 'next/router'
+import React from 'react'
+import { I18nextProvider, withSSR } from 'react-i18next'
+
+import i18n from './client'
+
+export default function withI18n(WrappedComponent: any) {
```

Comment:
I would need a second opinion here but I believe HOC should also have a display name so it's easier to debug 

```js
WrappedComponentWithSSR.displayName = `withI18n(${WrappedComponentWithSSR.displayName || WrappedComponentWithSSR.name})`
```